File: sast-python2/case/completeness/single_app_tracing/function_call/argument_passing/argument_passing_reference_001_F.py
# -*- coding: utf-8 -*-
# evaluation information start
# real case = false
# evaluation item = 完整度->单应用跟踪完整度->函数和方法调用->参数传递
# scene introduction = 引用传递
# level = 2
# bind_url = completeness/single_app_tracing/function_call/argument_passing/argument_passing_reference_001_F
# evaluation information end
import os


def argument_passing_reference_001_F(taint_src):
    obj = {u'data': u'_'}
    process(obj)
    taint_sink(obj[u'data'])


def process(obj):
    obj[u'data'] = u'_'


def taint_sink(o): 
    os.system(o)
    pass


if __name__ == u'__main__':
    taint_src = u"taint_src_value"
    argument_passing_reference_001_F(taint_src)

File: sast-go/cases/accuracy/context_sensitive/argument_return_value_passing/argument_passing_reference_001_F/argument_passing_reference_001_F.go

// evaluation information start
// real case = false
// evaluation item = 准确度->上下文敏感分析->参数/返回值传递
// scene introduction = 参数值传递->引用传递->map
// level = 2
// bind_url = accuracy/context_sensitive/argument_return_value_passing/argument_passing_reference_001_F/argument_passing_reference_001_F
// evaluation information end

package argument_passing_reference_001_F

func argument_passing_reference_001_F(__taint_src interface{}) {
	obj := map[string]interface{}{
		"data": __taint_src,
	}
	process(obj)
	__taint_sink(obj["data"])
}

func process(obj map[string]interface{}) {
	obj["data"] = "_"
}

func __taint_sink(o interface{}) {
}
